Scripts/script-UnPackFile.yml

```diff
@@ -0,0 +1,123 @@
+  import tempfile
+  from os.path import isfile, join
+  import subprocess
+
+  filePath = None
+  fileEntryID = ''
+  if demisto.args().has_key('fileName') or demisto.args().has_key('lastPackedFileInWarroom'):
+      entries = demisto.executeCommand('getEntries', {})
+      for entry in entries:
+          fn = demisto.get(entry, 'File')
+
+          is_text = type(fn) in [unicode, str]
+          is_correct_file = demisto.args().get('fileName', '').lower() == fn.lower()
+
+          if is_text:
+              if demisto.args().has_key('fileName') and is_correct_file:
+                  fileEntryID = entry['ID']
+                  break
+              if demisto.args().has_key('lastPackedFileInWarroom') and fn.lower().endswith(demisto.args().get('lastPackedFileInWarroom', '').lower()):
+                  fileEntryID = entry['ID']
+
+      if fileEntryID == '':
+          errorMessage = ''
+          if demisto.args().has_key('fileName'):
+              demisto.results({
+                  'Type': entryTypes['error'],
+                  'ContentsFormat': formats['text'],
+                  'Contents': '"' + demisto.args().get('fileName') + '" no such file in the war room'
+              })
+          if demisto.args().has_key('lastPackedFileInWarroom'):
+              demisto.results({
+                  'Type': entryTypes['error'],
+                  'ContentsFormat': formats['text'],
+                  'Contents': 'Could not find "' + demisto.args().get('lastPackedFileInWarroom', '') + '" file in war room'
+              })
+
+          sys.exit(0)
+
+  if demisto.args().has_key('entryID'):
+      fileEntryID = demisto.args().get('entryID')
+
+  if not fileEntryID:
+      demisto.results({
+          'Type': entryTypes['error'],
+          'ContentsFormat': formats['text'],
+          'Contents': 'You must set entryID or fileName or lastPackedFileInWarroom=i.e.(zip) when executing Unpack script'
+      })
+      sys.exit(0)
+
+  res = demisto.executeCommand('getFilePath', {'id': fileEntryID})
+  if res[0]['Type'] == entryTypes['error']:
+      demisto.results({
+          'Type': entryTypes['error'],
+          'ContentsFormat': formats['text'],
+          'Contents': 'Failed to get the file path for entry: ' + fileEntryID
+      })
+      sys.exit(0)
+
+  filePath = res[0]['Contents']['path']
+
+  filenames = []
+  tempdir = tempfile.mkdtemp(prefix='demisto_rar_')
+  Archive(filePath).extractall_patool(tempdir, None)
+  for root, directories, files in os.walk(tempdir):
+      for name in files:
+          if isfile(join(tempdir, name)):
+              filenames.append(name)
+
+  if len(filenames) == 0:
+      demisto.results({
+          'Type': entryTypes['error'],
+          'ContentsFormat': formats['text'],
+          'Contents': 'Could not find files in archive' + ','.join(os.listdir(tempdir))
+      })
+  else:
+      results = []
+      for filename in filenames:
+          ## Open extracted filename read and copy to warroom.
```

Comment:
I would change it to `file_result_existing_file` from common - you don't need to read the file into memory and write it again - just move it. See how UnzipFile now does it.